apps/backend/src/statusActionEngine/emailActionHandler.ts

```diff
@@ -21,203 +22,379 @@ import {
-          const FCSs =
-            await getFapChairSecretariesAndFormatOutputForEmailSending(
-              proposals,
-              recipientWithTemplate
-            );
-
-          sendMail(FCSs);
-
-          break;
-        }
-
-        case EmailStatusActionRecipients.USER_OFFICE: {
-          const adminDataSource = container.resolve<AdminDataSource>(
-            Tokens.AdminDataSource
-          );
-
-          const instrumentDataSource = container.resolve<InstrumentDataSource>(
-            Tokens.InstrumentDataSource
-          );
-
-          const userOfficeEmail = (
-            await adminDataSource.getSetting(SettingsId.USER_OFFICE_EMAIL)
-          )?.settingsValue;
-
-          if (!userOfficeEmail) {
-            logger.logError(
-              'Could not send email(s) to the User Office as the setting (USER_OFFICE_EMAIL) is not set.',
-              { proposalEmailsSkipped: proposals }
-            );
-
-            break;
-          }
-
-          let uoRecipient: EmailReadyType[];
-
-          if (recipientWithTemplate.combineEmails) {
-            uoRecipient = [
-              {
-                id: recipientWithTemplate.recipient.name,
-                email: userOfficeEmail,
-                proposals: proposals,
-                template: recipientWithTemplate.emailTemplate.id,
-              },
-            ];
-          } else {
-            const usersDataSource: UserDataSource = container.resolve(
-              Tokens.UserDataSource
-            );
-
-            const recipientPromises = proposals.map(async (proposal) => ({
-              id: recipientWithTemplate.recipient.name,
-              email: userOfficeEmail,
-              proposals: [proposal],
-              template: recipientWithTemplate.emailTemplate.id,
-              instruments:
-                await instrumentDataSource.getInstrumentsByProposalPk(
-                  proposal.primaryKey
-                ),
-              pi:
-                (await usersDataSource.getBasicUserInfo(proposal.proposerId)) ||
-                null,
-              coProposers:
-                (await usersDataSource.getProposalUsers(proposal.primaryKey)) ||
-                null,
-            }));
-
-            uoRecipient = await Promise.all(recipientPromises);
-          }
-
-          sendMail(uoRecipient);
-
-          break;
-        }
+  if (statusActionRecipients) {
+    const recipientWithTemplate = config.recipientsWithEmailTemplate.find(
+      (value) => value.recipient.name === statusActionRecipients
+    );
+
+    if (!recipientWithTemplate) {
+      throw new Error(
+        `Provided status action recipient does not match any available email recipients 
+        ${statusActionRecipients}`
+      );
+    }
+    emailStatusActionRecipients(
+      recipientWithTemplate,
+      proposalStatusAction,
+      proposals,
+      statusActionsLogId,
+      loggedInUserId,
+      recipientWithTemplate.recipient.name
```

Comment:
This is the redundant argument. `recipientWithTemplate` is already passed as the first parameter. Could we remove this?

Correct me if i am wrong.